### DISCUSSION TITLE
JIT with data attributes with variables

### DISCUSSION THREAD

**Discussion by minedun6** [tailwindlabs/tailwindcss#11785] [Help] [answered]
Is it possible to combine the data attributes styling with the JIT alongside variables ?

For example

```js
const color = '#0000ff'

<DropdownMenuPrimitive.Item
              key={`${label}-${i}`}
              className={clsx(
                "group flex cursor-default select-none items-center rounded-md px-2 py-2 text-xs outline-none",
                `text-gray-400 focus:bg-gray-50 data-[highlighted]:bg-${color} data-[highlighted]:text-${color}`
              )}
            >
              {icon}
              <span
                className={`flex-grow text-gray-700 group-data-[highlighted]:text-${color}`}
              >
                {label}
              </span>
              {shortcut && <span className="text-xs">{shortcut}</span>}
            </DropdownMenuPrimitive.Item>
```

Here's a link for better reproduction of the bug. [Sandbox](https://codesandbox.io/s/wizardly-wescoff-k864nx?file=/components/dropdown-menu.tsx)
Where is if I removed the color variable and replace it with the value #0000ff, it will work correctly

**Comment by wongjn** [comment, marked as answer]
As per [the documentation](https://tailwindcss.com/docs/content-configuration#dynamic-class-names):

> The most important implication of how Tailwind extracts class names is that it will only find classes that exist as _complete unbroken strings_ in your source files.
>
> If you use string interpolation or concatenate partial class names together, Tailwind will not find them and therefore will not generate the corresponding CSS:
>
> **Don’t construct class names dynamically**
>
> ```vue
> <div class="text-{{ error ? 'red' : 'green' }}-600"></div>
> ```
> In the example above, the strings `text-red-600` and `text-green-600` do not exist, so Tailwind will not generate those classes.
> Instead, make sure any class names you’re using exist in full:
>
> **Always use complete class names**
>
> ```vue
> <div class="{{ error ? 'text-red-600' : 'text-green-600' }}"></div>
> ``` 

Instead, consider using the `style` attribute for value:

```diff
   className={clsx(
     "group flex cursor-default select-none items-center rounded-md px-2 py-2 text-xs outline-none",
-    `text-gray-400 focus:bg-gray-50 data-[highlighted]:bg-${color} data-[highlighted]:text-${color}`
+    `text-gray-400 focus:bg-gray-50 data-[highlighted]:bg-[color:--color] data-[highlighted]:text-[color:--color]`
   )}
+  style={{ '--color': color }}
 >
```

Since you seem to be using Tailwind `v3.2`, we can't use the CSS variable shortcut syntax, so it would be:

```diff
   className={clsx(
     "group flex cursor-default select-none items-center rounded-md px-2 py-2 text-xs outline-none",
-    `text-gray-400 focus:bg-gray-50 data-[highlighted]:bg-${color} data-[highlighted]:text-${color}`
+    `text-gray-400 focus:bg-gray-50 data-[highlighted]:bg-[color:var(--color)] data-[highlighted]:text-[color:var(--color)]`
   )}
+  style={{ '--color': color }}
 >
```

[See forked Code Sandbox](https://codesandbox.io/s/https-github-com-tailwindlabs-tailwindcss-discussions-11785-tsj7kt?file=/components/dropdown-menu.tsx)

**Reply by minedun6** [reply]
Thank you for the answer.
Much appreciated for taking your time to answer with such details.

**Reply by karlhorky** [reply]
@adamwathan should this workaround with `style` be added to [the "Dynamic class names" docs section on the "Content Configuration" page](https://tailwindcss.com/docs/content-configuration#dynamic-class-names)?
